builtin/logical/mysql/secret_creds.go

```diff
@@ -82,22 +82,33 @@ func (b *backend) secretCredsRevoke(
 	}
 
 	// Get the role name
+	// we may not always have role data in the secret InternalData
+	// so don't exit if the roleNameRaw fails. Instead it is set
+	// as an empty string
+	var roleName string
 	roleNameRaw, ok := req.Secret.InternalData["role"]
-	roleName, ok := roleNameRaw.(string)
+	if !ok {
+		roleName = ""
+	} else {
+		roleName, ok = roleNameRaw.(string)
+	}
+	// init default revoke sql string.
+	// this will replaced by a user provided one if one exists
+	// otherwise this is what will be used when lease is revoked
+	revokeSQL := defaultRevokeSQL
 
-	// init role entry struct
-	var role = &roleEntry{}
+	// if we were successful in finding a role name
+	// create role entry from that name
 	if roleName != "" {
 		role, err = b.Role(req.Storage, roleName)
 		if err != nil {
 			return nil, err
 		}
```

Comment:
You should check for a nil `role` here -- if a role isn't found the role function returns `nil, nil`. In that case I'm not sure what the right thing to do would be -- probably to fall back to the default with a warning in the response to let the user know what happened.